### PR TITLE
chore(build): deploy core rules documentation to github pages

### PR DESCRIPTION
**Pull request by dileepkumarv-avio**
Mule Linter core is shipped with [many rules](https://github.com/avioconsulting/mule-linter/tree/main/mule-linter-core/src/main/groovy/com/avioconsulting/mule/linter/rule). When building the module, groovy docs are generated under ./mule-linter-core/build/docs. It will be helpful to publish these to github pages for reference.

This will resolve the issue with [github pages] deploy action (https://github.com/avioconsulting/mule-linter/issues/133)

### CODE REVIEW

File: .github/workflows/github-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  # Temporary on current branch
+  push:
+    branches:
+      - main
+      - feat/**
```
Comment (adesjardin) [inline .github/workflows/github-deploy.yml]: Don't publish pages on feature branches, only main

File: .github/workflows/github-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
```
Comment (adesjardin) [inline .github/workflows/github-deploy.yml:27]: Can we call this deploy-pages so its clear in actions what is being deployed?

File: .github/workflows/github-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
```
Comment (adesjardin) [inline .github/workflows/github-deploy.yml:38]: use latest version of the action, v3

File: .github/workflows/github-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Setup Pages
+        uses: actions/configure-pages@v2
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v1
+        with:
+          # Upload entire repository
+          path: './mule-linter-core/build/docs/groovydoc'
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v1
```
Comment (adesjardin) [inline .github/workflows/github-deploy.yml:54]: use latest version of the action (v2)

File: .github/workflows/github-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Verify
+        # build to generate docs folder
+        run: ./gradlew check
+      - name: Setup Pages
+        uses: actions/configure-pages@v2
```
Comment (adesjardin) [inline .github/workflows/github-deploy.yml:46]: use latest version (v3)

**Review by adesjardin — CHANGES_REQUESTED** [review]

File: .github/workflows/github-deploy.yml (hunk 2 above)
Comment (dileepkumarv-avio) [reply .github/workflows/github-deploy.yml]: rename job to deploy-pages

**Review by adesjardin — APPROVED** [review]

**Pull request merged**